### PR TITLE
PLAT-601-5 CurrentLocale Is Now Respected for BigDecimal Format

### PR DESCRIPTION
**Pull request by oliver-richers**
- PLAT-601-5 Extended LocaleValidator
- PLAT-601-5 DomBigDecimalInput Now Respects Locale
- PLAT-601-5 EmfBigDecimalDisplay Now Respects Locale
- PLAT-601-5 Removed Useless BigDecimal Related Classes
- PLAT-601-5 EmfDataTableBigDecimalCellBuilder Now Respects Locale
- PLAT-601-5 I18nKeyDisplayString Now Respects Locale

### CODE REVIEW

File: platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.softicar.platform.common.core.locale.CurrentLocale;
+import com.softicar.platform.common.core.number.formatter.BigDecimalFormatter;
+import java.math.BigDecimal;
+
+/**
+ * Transforms I18n formatting arguments according to {@link CurrentLocale}.
+ *
+ * @author Oliver Richers
+ */
+class I18nFormatArgumentsProcessor {
```
Comment (alexander-schmidt) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java:12]:
```suggestion
class I18nFormatArgumentsLocalizer {
```
This class is all about localization. We could (and probably should) express that in its name.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-common-core/src/test/java/com/softicar/platform/common/core/locale/LocaleValidatorTest.java [hunk 2]
```diff
@@ -1,44 +1,69 @@
...
-	@Test(expected = SofticarUserException.class)
-	public void testDigitGroupSeparatorWithDigits() {
+	private static void assertValidationException(IDisplayString expectedMessage, Locale locale) {
 
-		Locale locale = new Locale()//
-			.setDigitGroupSeparator("123");
+		assertException(expectedMessage, () -> new LocaleValidator(locale).validate());
+	}
+
+	private static void assertException(IDisplayString expectedMessage, INullaryVoidFunction thrower) {
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common-core/src/test/java/com/softicar/platform/common/core/locale/LocaleValidatorTest.java:60]: tja ;)

File: platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java [hunk 3]
```diff
@@ -17,6 +18,10 @@
 				.de("Das Dezimaltrennzeichen muss sich vom Zifferngruppentrennzeichen unterscheiden.");
 	I18n0 THE_DECIMAL_SEPARATOR_MUST_NOT_CONTAIN_DIGITS = new I18n0("The decimal separator must not contain digits.")//
 		.de("Das Dezimaltrennzeichen darf keine Ziffern beinhalten.");
+	I18n1 THE_DECIMAL_SEPARATOR_MUST_NOT_CONTAIN_THESE_CHARACTERS_ARG1 = new I18n1("The decimal separator must not contain these characters: %s")//
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java:21]: ok

File: platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	private final Object[] arguments;
+
+	public I18nFormatArgumentsProcessor(Object[] arguments) {
+
+		this.arguments = arguments;
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java:18]: ok

File: platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java (hunk 1 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/i18n/I18nFormatArgumentsProcessor.java:12]: ok, renamed

File: platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java [hunk 5]
```diff
@@ -5,6 +5,7 @@
 
 public class LocaleValidator {
 
+	private final static String ILLEGAL_CHARACTERS = "+-eE";
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/locale/LocaleValidator.java:8]: renamed to `ILLEGAL_SEPARATOR_CHARACTERS`

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**